solidity/test/token_stake/TestStakingPortBacker.js

```diff
@@ -0,0 +1,635 @@
+        ),
+        "Corrupted input data"
+      )    
+    })
+
+    it("can be done only one time for the operator", async () => {
+      const data = web3.eth.abi.encodeParameters(["address"], [operatorOne])
+      await token.approveAndCall(
+        stakingPortBacker.address, delegatedAmount, data, {from: tokenOwner}
+      )  
+      await expectRevert(
+        token.approveAndCall(
+          stakingPortBacker.address, delegatedAmount, data, {from: tokenOwner}
+        ),
+        "Already paid back"
+      ) 
+    })
+
+    it("changes liquid tokens staking relationship owner to token owner", async () => {
+      const data = web3.eth.abi.encodeParameters(["address"], [operatorOne])
+      await token.approveAndCall(
+        stakingPortBacker.address, delegatedAmount, data, {from: tokenOwner}
+      )
+      
+      expect(await newTokenStaking.ownerOf(operatorOne)).to.equal(tokenOwner)
+    })
+
+    it("changes grant staking relationship owner to grantee", async () => {
+      const data = web3.eth.abi.encodeParameters(["address"], [operatorTwo])
+      await token.transfer(grantee, delegatedAmount, {from: deployer})
+      await token.approveAndCall(
+        stakingPortBacker.address, delegatedAmount, data, {from: grantee}
+      )
+      
+      expect(await newTokenStaking.ownerOf(operatorTwo)).to.equal(grantee) 
+    })
+
+    it("changes managed grant staking relationship owner to managed grantee", async () => {
+      const data = web3.eth.abi.encodeParameters(["address"], [operatorThree])
+      await token.transfer(managedGrantee, delegatedAmount, {from: deployer})
+      await token.approveAndCall(
+        stakingPortBacker.address, delegatedAmount, data, {from: managedGrantee}
+      )
+      
+      expect(await newTokenStaking.ownerOf(operatorThree)).to.equal(managedGrantee) 
+    })
+
+    it("fails when not done by the eventual staking relationship owner", async () => {
+      const data = web3.eth.abi.encodeParameters(["address"], [operatorOne])
+      await expectRevert(
+        token.approveAndCall(
+          stakingPortBacker.address, delegatedAmount, data, {from: operatorOne}
+        ),
+        "Not authorized to pay back"
+      )
+      await expectRevert(
+        token.approveAndCall(
+          stakingPortBacker.address, delegatedAmount, data, {from: thirdParty}
+        ),
+        "Not authorized to pay back"
+      )
+    })
+  })
+
+  describe("withdraw", async () => {
+    it("fails when not called by the owner", async () => {
+      await expectRevert(
+        stakingPortBacker.withdraw(1000, {from: thirdParty}),
+        "Ownable: caller is not the owner"
+      )
+    })
+
+    it("allows owner to withdraw tokens", async () => {
+      const balanceBefore = await token.balanceOf(owner)
+      await stakingPortBacker.withdraw(9999, {from: owner})
+      const balanceAfter = await token.balanceOf(owner)
+
+      expect(balanceAfter.sub(balanceBefore)).to.eq.BN(9999)
+    })
```

Comment:
Do we want to add a case for withdraw where the balance is insufficient?

Reply:
Hmm... I don't know, it seems to be covered by ERC20 implementation, right?